lighthouse-core/gather/computed/page-dependency-graph.js

```diff
@@ -0,0 +1,122 @@
+  get name() {
+    return 'PageDependencyGraph';
+  }
+
+  get requiredNumberOfArtifacts() {
+    return 2;
+  }
+
+  /**
+   * @param {!WebInspector.NetworkRequest} record
+   * @return {!Array<string>}
+   */
+  static getNetworkInitiators(record) {
+    if (!record._initiator) return [];
+    if (record._initiator.url) return [record._initiator.url];
+    if (record._initiator.type === 'script') {
+      const frames = record._initiator.stack.callFrames;
+      return Array.from(new Set(frames.map(frame => frame.url))).filter(Boolean);
+    }
+
+    return [];
+  }
+
+  /**
+   * @param {!TraceOfTabArtifact} traceOfTab
+   * @param {!Array<!WebInspector.NetworkRequest>} networkRecords
+   * @return {!Node}
+   */
+  static createGraph(traceOfTab, networkRecords) {
+    const idToNodeMap = new Map();
+    const urlToNodeMap = new Map();
+
+    networkRecords.forEach(record => {
+      const node = new Node(record.requestId);
+      idToNodeMap.set(record.requestId, node);
+
+      if (urlToNodeMap.has(record.url)) {
+        // If duplicate requests have been made to this URL we can't be certain which node is being
+        // referenced, so act like we don't know the URL at all.
+        urlToNodeMap.set(record.url, undefined);
+      } else {
+        urlToNodeMap.set(record.url, node);
+      }
+    });
+
+    const rootRequest = networkRecords
+        .reduce((min, next) => min.startTime < next.startTime ? min : next);
+    const rootNode = idToNodeMap.get(rootRequest.requestId);
+    networkRecords.forEach(record => {
+      const initiators = PageDependencyGraphArtifact.getNetworkInitiators(record);
+      const node = idToNodeMap.get(record.requestId);
+      if (initiators.length) {
+        initiators.forEach(initiator => {
+          const parent = urlToNodeMap.get(initiator) || rootNode;
+          parent.addDependent(node);
+        });
+      } else if (record !== rootRequest) {
+        rootNode.addDependent(node);
+      }
+    });
+
+    return rootNode;
+  }
+
+  /**
+   * @param {!Node} rootNode
+   * @return {number}
+   */
+  static computeGraphDuration(rootNode) {
+    const depthByNodeId = new Map();
+    const getMax = arr => Array.from(arr).reduce((max, next) => Math.max(max, next), 0);
+
+    let startingMax = Infinity;
+    let endingMax = Infinity;
+    while (endingMax === Infinity || startingMax > endingMax) {
+      startingMax = endingMax;
+      endingMax = 0;
+
+      rootNode.traverse(node => {
+        const dependencies = node.getDependencies();
+        const dependencyDepths = dependencies.map(node => depthByNodeId.get(node.id) || Infinity);
+        const maxDepth = getMax(dependencyDepths);
+        endingMax = Math.max(endingMax, maxDepth);
+        depthByNodeId.set(node.id, maxDepth + 1);
+      });
+    }
+
+    const maxDepth = getMax(depthByNodeId.values());
+    return maxDepth * Emulation.settings.TYPICAL_MOBILE_THROTTLING_METRICS.latency;
```

Comment:
Structure-wise I feel like it'd be a tad better to have this be a generic graph `computeMaxDepth` method. 

Optionally, I could see this latency work going into the audit rather than here in the graph computed artifact.

You are now telling me verbally that this all disappears in the next commit... so.... 💨 💨 💨 💨